test/testing/test_pytest_plugin.py

```diff
@@ -136,6 +136,113 @@ def test_container_factory(container_factory, rabbit_config):
     assert get_rabbit_connections(vhost, rabbit_manager) == []
 
 
+def test_container_factory_with_custom_container_cls(testdir):
+
+    testdir.makeconftest(
```

Comment:
as opposed to a dynamically generated one? seems nicer to have it all contained within this test, rather than having a file floating around somewhere else.